File: app/src/main/java/com/plweegie/android/squash/receivers/StartupReceiver.java
package com.plweegie.android.squash.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.plweegie.android.squash.utils.WorkManagerUtil;

public class StartupReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {

        if (intent.getAction().equals(Intent.ACTION_BOOT_COMPLETED)) {
            WorkManagerUtil.enqueueWorkRequest();
        }
    }
}

File: app/src/main/java/com/plweegie/android/squash/receivers/StartupReceiver.kt
package com.plweegie.android.squash.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.work.ExistingPeriodicWorkPolicy
import com.crashlytics.android.Crashlytics
import com.plweegie.android.squash.utils.WorkManagerUtil


class StartupReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context?, intent: Intent?) {
        if (intent?.action == Intent.ACTION_BOOT_COMPLETED && context != null) {
            Crashlytics.log("Boot action received")
            WorkManagerUtil.enqueueWorkRequest(context,
                    policy = ExistingPeriodicWorkPolicy.REPLACE)
        }
    }
}